app/controllers/shared/api.php

```diff
@@ -114,6 +114,36 @@
 
     $deletes->setProject($project);
     $database->setProject($project);
+
+    $useCache = $route->getLabel('cache', false);
+    if ($useCache) {
+        $key = md5($request->getURI() . implode('*', $request->getParams()));
+        $cache = new Cache(new Filesystem(APP_STORAGE_CACHE . DIRECTORY_SEPARATOR . 'app-' . $project->getId()));
+        $timestamp = 60 * 60 * 24 * 30;
+        $data = $cache->load($key, $timestamp);
+        if (!empty($data)) {
+            $cacheLog = Authorization::skip(fn () => $dbForProject->getDocument('cache', $key));
+            if ($cacheLog->isEmpty()) {
+                Authorization::skip(fn () => $dbForProject->createDocument('cache', new Document([
+                '$id' => $key,
+                'accessedAt' => time(),
+                ])));
+            } else {
+                $cacheLog->setAttribute('accessedAt', time());
```

Comment:
We need to update this attribute only once a day not on every request, this could overload the DB with write requests and significantly slow down our API.